python-sdk/src/astro/databases/snowflake.py

```diff
@@ -579,17 +567,20 @@ def load_pandas_dataframe_to_table(
         :param if_exists: Strategy to be used in case the target table already exists.
         :param chunk_size: Specify the number of rows in each batch to be written at a time.
         """
-        if if_exists == "replace":
-            self.create_table(target_table, dataframe=source_dataframe)
+
+        auto_create_table = False
+        if if_exists == "replace" or not self.table_exists(target_table):
+            auto_create_table = True
 
         pandas_tools.write_pandas(
             conn=self.hook.get_conn(),
             df=source_dataframe,
-            table_name=target_table.name,
+            table_name=target_table.name.upper(),
```

Comment:
Why is this set to upper now?

Reply:
Added a response here - https://github.com/astronomer/astro-sdk/pull/1050#discussion_r995468099

Reply:
Can you add code comments about this please?

Reply:
@utkarsharma2 Can you address this please

Reply:
This needs to be added as comment

Reply:
Done.